client/clparser/clparser.c

```diff
@@ -0,0 +1,163 @@
+//-----------------------------------------------------------------------------
+// Copyright (C) 2017 Merlok
+//
+// This code is licensed to you under the terms of the GNU GPL, version 2 or,
+// at your option, any later version. See the LICENSE.txt file for the text of
+// the license.
+//-----------------------------------------------------------------------------
+// Command line parser core commands
+//-----------------------------------------------------------------------------
+
+#include "clparser.h"
```

Comment:
Please consider calling this `cliparser` (and the methods `CLI*`), as this is a more familiar acronym.  `CL` has other, different meanings (Wikipedia snippet), none of which are "command line".

```
Computing[edit]
CAS latency, a measure used in computer memory
Combinatory logic, in computer science
Common Lisp, a programming language
Common Logic, a framework for a family of logic languages
AS/400 Control Language, a scripting language for the IBM AS/400 midrange platform
OpenCL, a programming language for heterogeneous platforms
CL register, the low byte of an X86 16-bit CX register
cl.exe, the command-line C/C++ compiler for Microsoft Visual C++
Internet[edit]
.cl, Internet country code top-level domain for Chile
Chess Live, an Internet chess server
craigslist, an online classifieds service
```

In Perforce, then CL means "Change List".  As someone who uses Perforce regularly, that was the first thing that came to mind.